### PR TITLE
Sub/refactor/68 split project members from projects reducer

### PR DESCRIPTION
**Pull request by tanmaythole**
Move the project members' reducer from the projects reducer to the separate reducers file.

### CODE REVIEW

File: src/containers/DashBoard/index.jsx [hunk 1]
```diff
@@ -1,76 +1,75 @@
...
-} from 'Components/DashboardComponent';
-import { DashboardLayout } from 'Layout';
-import { addProject, fetchProjects } from 'store/Projects/actions';
+import {
+  AddProjectModal,
+  DashboardSubHeader,
+  ProjectsComponent,
+} from "Components/DashboardComponent";
+import { DashboardLayout } from "Layout";
+import { addProjectRequest, getProjectsRequest } from "store/Projects/actions";
```
Comment (ChinmaySonawane) [inline src/containers/DashBoard/index.jsx:11]: need empty line

File: src/containers/DashBoard/index.jsx [hunk 2]
```diff
@@ -1,76 +1,75 @@
...
+  };
 
-    const onchangeProjectFormData = (e) => {
-        setAddProjectFormData(p => ({
-            ...p,
-            [e.target.name]: e.target.value
-        }))
-    }
+  const onchangeProjectFormData = (e) => {
+    setAddProjectFormData((p) => ({
```
Comment (ChinmaySonawane) [inline src/containers/DashBoard/index.jsx]: instead of p we can use prev or prevState

File: src/containers/DashBoard/index.jsx [hunk 3]
```diff
@@ -1,76 +1,75 @@
...
+    setShowAddProjectModal(!showAddProjectModal);
+  };
 
-    const onchangeProjectFormData = (e) => {
-        setAddProjectFormData(p => ({
-            ...p,
-            [e.target.name]: e.target.value
-        }))
-    }
+  const onchangeProjectFormData = (e) => {
```
Comment (ChinmaySonawane) [inline src/containers/DashBoard/index.jsx]: rename this to `onChangeProjectFormData`

File: src/store/ProjectMembers/reducers.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import projectMembersConstants from "./constants";
+
+const initialState = {
+  project: "",
+  members: [],
+  project_admin: null,
```
Comment (ChinmaySonawane) [inline src/store/ProjectMembers/reducers.js]: rename this to `projectAdmin`

File: src/store/ProjectMembers/sagas.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  addMembersInProjectFailure,
+  getProjectMembersFailure,
+  getProjectMembersSuccess,
+  removeMembersInProjectFailure,
+} from "./actions";
+
+import { MEMBERS } from "constants/userMessagesConstants";
+import projectMembersConstants from "./constants";
+
+import { apisErrorMessage } from "utils/apisErrorMessage";
```
Comment (ChinmaySonawane) [inline src/store/ProjectMembers/sagas.js]: helper and utils should be over constants

File: src/store/ProjectMembers/sagas.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    toastMessage(errorMessage, "error");
+    yield put(getProjectMembersFailure());
+  }
+}
+
+export function* addMembersInProject({ payload }) {
+  try {
+    yield call(addMembersInProjectApi, payload);
+    yield call(getProjectMembers, { payload: { project: payload.project } });
+    toastMessage(MEMBERS.ADD_NEW_SUCCESS, "success");
```
Comment (ChinmaySonawane) [inline src/store/ProjectMembers/sagas.js]: no need to pass "success" as we kept it as a default value

File: src/store/Projects/actions.js [hunk 7]
```diff
@@ -1,106 +1,67 @@
 import projectConstants from "./constants";
```
Comment (ChinmaySonawane) [inline src/store/Projects/actions.js:1]: Lets discuss about naming convention

File: src/containers/DashBoard/index.jsx (hunk 1 above)
Comment (tanmaythole) [reply src/containers/DashBoard/index.jsx:11]: Done 👍🏻

File: src/containers/DashBoard/index.jsx (hunk 2 above)
Comment (tanmaythole) [reply src/containers/DashBoard/index.jsx]: Done 👍🏻

File: src/containers/DashBoard/index.jsx (hunk 3 above)
Comment (tanmaythole) [reply src/containers/DashBoard/index.jsx]: Done 👍🏻

File: src/store/ProjectMembers/reducers.js (hunk 4 above)
Comment (tanmaythole) [reply src/store/ProjectMembers/reducers.js]: Done 👍🏻

File: src/store/ProjectMembers/sagas.js (hunk 5 above)
Comment (tanmaythole) [reply src/store/ProjectMembers/sagas.js]: Done 👍🏻

File: src/store/ProjectMembers/sagas.js (hunk 6 above)
Comment (tanmaythole) [reply src/store/ProjectMembers/sagas.js]: Done 👍🏻

File: src/store/Projects/actions.js (hunk 7 above)
Comment (tanmaythole) [reply src/store/Projects/actions.js:1]: Done 👍🏻

File: src/containers/ProjectContainer/SettingsContainer/index.jsx [hunk 8]
```diff
@@ -1,101 +1,112 @@
...
+
+  return (
+    <div className="px-5 py-4 w-100">
+      <div className={`py-3 d-flex justify-content-between`}>
+        <div>
+          <div className="text-primary">
+            <span
+              className="d-flex align-items-center"
+              style={{ cursor: "pointer", width: "fit-content" }}
+              onClick={() => navigate(-1)}
```
Comment (ChinmaySonawane) [inline src/containers/ProjectContainer/SettingsContainer/index.jsx:83]: what is `navigate(-1)`?

File: src/store/Projects/apis.js [hunk 9]
```diff
@@ -1,40 +1,16 @@
...
-import { Api } from "../apiHelper";
+import { Api } from "store/apiHelper";
 
+import { DELETE, GET, POST, SERVER_URL } from "constants/appConstants";
 
-export const getProjectsApi = () => {
-    return Api(`${SERVER_URL}/api/projects/`, GET)
-}
+export const getProjectsApi = () => Api(`${SERVER_URL}/api/projects/`, GET);
+export const addNewProjectApi = (data) =>
```
Comment (ChinmaySonawane) [inline src/store/Projects/apis.js:7]: can we add empty line between two constants

File: src/containers/ProjectContainer/SettingsContainer/index.jsx (hunk 8 above)
Comment (tanmaythole) [reply src/containers/ProjectContainer/SettingsContainer/index.jsx:83]: `navigate(-1)` is used to go back to the previous route

File: src/store/Projects/apis.js (hunk 9 above)
Comment (tanmaythole) [reply src/store/Projects/apis.js:7]: Sure 👍🏻

**Review by ChinmaySonawane — APPROVED** [review]

**Pull request merged**